src/TemporaryUploadedFile.php

```diff
@@ -142,20 +142,51 @@ public static function createFromLivewire($filePath)
 
     public static function canUnserialize($subject)
     {
-        if (! is_string($subject)) return false;
+        if (is_string($subject)) {
+            return Str::startsWith($subject, 'livewire-file:')
+                || Str::startsWith($subject, 'livewire-files:');
+        } elseif (is_array($subject)) {
+            return collect($subject)->filter(function ($key) {
+                return ! empty($key);
+            })->contains(function ($value) {
+                if (is_string($value)) {
```

Comment:
This should be recursive so you aren't repeating yourself, AND so it covers more than 2 layers deep.

Reply:
Here is the recursive code from my PR.

```suggestion
            return collect($subject)->filter(function ($key) {
                return ! empty($key);
            })->contains(function ($value) {
                return static::canUnserialize($value);
            });
```